### PR TITLE
feat(dpg): improve client summary

### PR DESCRIPTION
**Pull request by archerzz**
part of Azure/autorest.csharp#2195# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by archerzz** [comment]
proceed https://github.com/Azure/azure-sdk-for-net/pull/29041

File: sdk/purview/Azure.Analytics.Purview.Account/src/Generated/PurviewCollection.cs [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
 namespace Azure.Analytics.Purview.Account
 {
+    // Data plane generated client. The PurviewCollection service client.
```
Comment (annelo-msft) [inline sdk/purview/Azure.Analytics.Purview.Account/src/Generated/PurviewCollection.cs:19]: Would it be hard to indicate the difference between service clients and subclients?

Comment (archerzz) [reply sdk/purview/Azure.Analytics.Purview.Account/src/Generated/PurviewCollection.cs]: @annelo-msft No. It's straightforward. I've updated the generator to print `sub-client` if the client is a sub-client. See the latest changes in this PR.

Comment (annelo-msft) [reply sdk/purview/Azure.Analytics.Purview.Account/src/Generated/PurviewCollection.cs]: Ok, great, thanks @archerzz!

File: sdk/purview/Azure.Analytics.Purview.Account/src/Generated/PurviewResourceSetRule.cs [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 namespace Azure.Analytics.Purview.Account
 {
+    // Data plane generated sub-client. The PurviewResourceSetRule service client.
```
Comment (annelo-msft) [inline sdk/purview/Azure.Analytics.Purview.Account/src/Generated/PurviewResourceSetRule.cs:16]: We shouldn't call it a service client in the comment if it is a sub-client.  A service client is a top-level client (e.g. not a subclient, see: https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-subclients)

Comment (archerzz) [reply sdk/purview/Azure.Analytics.Purview.Account/src/Generated/PurviewResourceSetRule.cs]: @annelo-msft Thanks for pointing it out. I've updated the description of sub-clients. See the latest change in this PR.

**Review by annelo-msft — APPROVED** [review]
This LGTM now.  You may want to ask @mikekistler if he's satisfied with this for the purposes of identifying which services have DPG clients.  Thanks!

**Review by mikekistler — APPROVED** [review]
This looks good to me. I can just search the repo for "// Data plane generated cleint" and they will all pop out. Just what I need. Thx.

**Comment by archerzz** [comment]
resolved in PR #30902

**Pull request closed without merging**